### ISSUE TITLE
Restructuring this package to fit newer python packaging standard

### ISSUE THREAD

**Issue by GalBenZvi** [nipreps/eddymotion#63]
I think it may be valuable to restructure this package to [a newer standard](https://packaging.python.org/en/latest/tutorials/packaging-projects/).
It should be relatively easy to do and will establish a "cleaner" structure.

Using [some cookie-cutter template](https://github.com/ionelmc/cookiecutter-pylibrary) may do the trick, but I would love to get your input on this suggestion before I start working on it :satisfied:

Thanks in advance!

**Comment by oesteban** [comment]
What is particularly missing?

**Comment by GalBenZvi** [comment]
Nothing very crucial, buy:

- The source code should be located under a folder called "src".
- Tests should be separated from the source code ("tests" folder).
- Setup configuration for the package should be located at "setup.cfg"
There may be some minor structural changes. I've [already started working](https://github.com/GalBenZvi/eddymotion/tree/structure) on this and am actually almost done with setting the structure.

It shouldn't affect the functionality in any way ofcourse.

**Comment by oesteban** [comment]
> * The source code should be located under a folder called "src".

Is this mandated by any PEP? (cc @effigies)

> * Tests should be separated from the source code ("tests" folder).

As above, I think this is just a suggestion. Personally, I find it easier to have tests with the code, and makes testing without installing easier.

> * Setup configuration for the package should be located at "setup.cfg"

Isn't this already this way?

>   There may be some minor structural changes. I've [already started working](https://github.com/GalBenZvi/eddymotion/tree/structure) on this and am actually almost done with setting the structure.

Happy to accept suggestions, but honestly not very open to set this very high priorities compared to other issues in the roadmap.

**Comment by effigies** [comment]
The packaging standards are:

* [PEP 517](https://peps.python.org/pep-0517/) (with the related, but irrelevant-to-us [PEP 518](https://peps.python.org/pep-0518))
* [PEP 621](https://peps.python.org/pep-0621/)
* Honorable mention: [PEP 420](https://peps.python.org/pep-0420/)

The PyPA tutorials are guides, not standards and are intended to get people going. They're not bad, but they're also frequently out of date with up-and-coming best practices. Until they recommend PEP-517 and PEP-621 (standardizing on pyproject.toml with no setup.py/setup.cfg) I will see them as transitional.

To this point, there is an extensive discussion currently ongoing about rewriting this tutorial not to be so tightly tied to one backend like setuptools:
https://discuss.python.org/t/removing-setup-cfg-and-setup-py-from-the-packaging-tutorial/16096/

**Comment by GalBenZvi** [comment]
I see... Thanks for the clarification @effigies!

In any way, if you would like to consider following the guides offered by PyPA, [I've finished restructuring the project](https://github.com/GalBenZvi/eddymotion/tree/restructure).

It includes mainly the following changes:
- Staging the source code under src/ rather than the package's base.
- Tests are also located there (in proximity to the source code, as suggested by @oesteban)
- Setup configurations are mostly the same as the original package.
- I've added [Github Action](https://github.com/GalBenZvi/eddymotion/blob/restructure/.github/workflows/github-actions.yml) and [pre-commit hooks](https://github.com/GalBenZvi/eddymotion/blob/restructure/.pre-commit-config.yaml) so that it'll be easier for you to control incoming suggestions, especially in light of the coming Hackathon at the OHBM.
- Added sphinx-appropriate documentation structure (docs folder and its derivatives).

I of course made sure that no original functionality was damaged due to these suggestions. Most tests are passing and I think those that do not also fail for the original structure (i.e, it's not a result of the restructuring).

Anyway, thank you for responding to my suggestions. :smile:
